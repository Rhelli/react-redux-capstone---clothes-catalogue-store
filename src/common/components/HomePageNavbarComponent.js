import React from 'react';

const HomePageNavbarComponent = () => (
  <div>
    <div>
      <h1>Clothes Inc.</h1>
    </div>
  </div>
);

export default HomePageNavbarComponent;
