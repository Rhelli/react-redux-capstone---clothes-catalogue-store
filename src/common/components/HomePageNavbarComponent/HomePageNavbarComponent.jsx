import React from 'react';
import styles from './HomePageNavbarComponent.module.scss';

const HomePageNavbarComponent = () => (
  <div className={styles.navbarContainer}>
    <div className={styles.navbarTitle}>
      <h1>Clothes Inc.</h1>
    </div>
  </div>
);

export default HomePageNavbarComponent;
