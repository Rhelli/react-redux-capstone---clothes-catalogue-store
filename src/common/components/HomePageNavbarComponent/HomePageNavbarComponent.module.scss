.navbarContainer {
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: #fdf2e9;
  height: 60px;
  border-bottom: 1px solid #ddd4cc;
  -webkit-box-shadow: 0 6px 5px -4px rgba(158, 151, 146, 1);
  -moz-box-shadow: 0 6px 5px -4px rgba(158, 151, 146, 1);
  box-shadow: 0 6px 5px -4px rgba(158, 151, 146, 1);
}

.navbarTitle {
  font: 700 20px/20px 'Playfair Display', sans-serif;
  color: #303030;
  padding: 12px 16px;
}
