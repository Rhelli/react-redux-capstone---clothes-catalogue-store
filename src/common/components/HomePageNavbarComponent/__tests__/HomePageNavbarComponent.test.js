import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import HomePageNavbarComponent from '../HomePageNavbarComponent';

describe('The appearance of the HomePageNavbarComponent', () => {
  it('Matches the HomePageNavbarComponent snapshot', () => {
    const navbarSnapshot = renderer.create(<HomePageNavbarComponent />).toJSON();
    expect(navbarSnapshot).toMatchSnapshot();
  });

  it('Has only one element, being the title branding', () => {
    const { container, getByText } = render(<HomePageNavbarComponent />);
    expect(container.firstChild.firstChild.firstChild.nodeName).toBe('H1');
    getByText('Clothes Inc.');
  });

  it('Has a two dimensional container structure', () => {
    const { container } = render(<HomePageNavbarComponent />);
    expect(container.firstChild.classList.contains('navbarContainer')).toBe(true);
    expect(container.firstChild.firstChild.classList.contains('navbarTitle')).toBe(true);
  });
});
