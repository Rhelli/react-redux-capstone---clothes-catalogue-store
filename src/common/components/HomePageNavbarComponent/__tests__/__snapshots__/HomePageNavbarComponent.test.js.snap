// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The appearance of the HomePageNavbarComponent Matches the HomePageNavbarComponent snapshot 1`] = `
<div
  className="navbarContainer"
>
  <div
    className="navbarTitle"
  >
    <h1>
      Clothes Inc.
    </h1>
  </div>
</div>
`;
